src/PAModel/Utility/DirectoryHelper.cs

```diff
@@ -71,7 +71,7 @@ public void WriteAllXML(string subdir, FilePath filename, string xmlText)
 
         public void WriteAllText(string subdir, FilePath filename, string text)
         {
-            string path = Path.Combine(_directory, subdir, filename.ToPlatformPath());
+            string path = FilePath.ToValidPath(Path.Combine(_directory, subdir, filename.ToPlatformPath()));
```

Comment:
Oh, I see. you're truncating the whole path, not just the part we're responsible for

Reply:
What if the base + _subdir part of the path exceeds MaxPath. Is that possible?

Reply:
yea I was thinking that, but I don't think thats possible, if that happens we won't be able to unpack anyways